ocm/__main__.py

```diff
@@ -0,0 +1,229 @@
+import argparse
+import dataclasses
+import datetime
+import json
+import os
+import sys
+
+import ocm
+
+
+try:
+    import yaml
+    _have_yaml = True
+except ImportError:
+    _have_yaml = False
+
+try:
+    import oci
+    import oci.auth
+    import oci.client
+    _have_oci = True
+except ImportError:
+    _have_oci = False
+
+
+def _parse_yaml_or_json(path) -> dict:
+    with open(path) as f:
+        if _have_yaml:
+            raw = yaml.safe_load(f)
+        else:
+            raw = json.load(f)
+
+    return raw
+
+
+def dump(component_descriptor: ocm.ComponentDescriptor, parsed):
+    if parsed.out == '-':
+        outfh = sys.stdout
+    else:
+        outfh = open(parsed.out, 'w')
+
+    raw_dict = dataclasses.asdict(component_descriptor)
+
+    if _have_yaml:
```

Comment:
Again, maybe we should add a `--json` and `--yaml` flag and determine the default based on the presence of the `yaml` module?

Reply:
yes, we could. do you see an actual use-case where someone would want to force usage of json, even if yaml module is present?